test/e2e/predictor/test_vllm.py

```diff
@@ -0,0 +1,745 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import os
+import ast
+
+import pytest
+from kubernetes import client
+from kubernetes.client import V1ResourceRequirements
+import numpy as np
+
+from kserve import (
+    V1beta1PredictorSpec,
+    V1beta1ModelSpec,
+    V1beta1ModelFormat,
+    V1beta1InferenceService,
+    V1beta1InferenceServiceSpec,
+    KServeClient,
+)
+from kserve.constants import constants
+from ..common.utils import (
+    KSERVE_TEST_NAMESPACE,
+    generate,
+    embed,
+    predict_isvc,
+    chat_completion_stream,
+    completion_stream,
+)
+from .test_output import (
+    huggingface_text_embedding_expected_output,
+    huggingface_sequence_classification_with_raw_logits_expected_output,
+)
+
+from kserve.logging import trace_logger
+
+
+@pytest.mark.llm
+def test_vllm_openai_chat_completions():
+    service_name = "vllm-qwen-chat"
+    predictor = V1beta1PredictorSpec(
+        min_replicas=1,
+        model=V1beta1ModelSpec(
+            model_format=V1beta1ModelFormat(
+                name="vllm",
+            ),
+            env=[
+                client.V1EnvVar(
+                    name="MODEL_ID",
+                    value="Qwen/Qwen2-0.5B-Instruct",
```

Comment:
This model size is around 1.2 GB. Having to download it for every test is not a good thing. Can we use the tiniest model available for VLLM?